### ISSUE TITLE
Results Reproduce

### ISSUE THREAD

**Issue by RAIAN08** [bjzhb666/GS-LoRA#10]
This is an excellent piece of work, but I am unable to reproduce the results presented in the paper. After three days of communication with the author and repeatedly modifying the code, I still cannot achieve the same results. Below, I list the modifications and process I followed. I sincerely suggest that the author recheck the open-source code.

## An issue related to the model structure

Below are the logs of the training process using `./scripts/run_cl_forget.sh` and the logs released by the author. In the highlighted area, a difference in the model structure can be observed. 
![fig1](https://github.com/user-attachments/assets/c8d9518c-f071-41aa-be40-a354bff997d8)

After discussing with the author, it was confirmed that `lora.MergedLinear` should be replaced with `nn.Linear`.

The complete code modification is as follows: `./vit_pytorch_face/vit_face.py`.
```
class Attention(nn.Module):
    def __init__(self, dim, heads=8, dim_head=64, dropout=0.0, lora_rank=8):
        super().__init__()
        inner_dim = dim_head * heads
        self.heads = heads
        self.scale = dim**-0.5

        print('in Attention class, lora_rank is:{}'.format(lora_rank))
        # self.to_qkv = nn.Linear(dim, inner_dim * 3, bias = False)
        # self.to_qkv = lora.MergedLinear(
        #     in_features=dim, #  latent dim = 512
        #     out_features=inner_dim * 3, # inner_dim == latent_dim
        #     r=lora_rank,
        #     enable_lora=[True, True, True],
        #     bias=False,
        # )
        self.to_qkv = nn.Linear(dim, inner_dim * 3, bias = False)
        self.to_out = nn.Sequential(nn.Linear(inner_dim, dim), nn.Dropout(dropout))
```
The same issue has been raised by the author here: https://github.com/bjzhb666/GS-LoRA/issues/9#issue-2512792028

## Anomalies in the ACC_o Metric

After making the above modifications, I retrained the model but still could not obtain results even remotely similar to those reported in the paper, especially regarding the Acc_o metric, as shown in Table 3 of the main paper. Below is my training log.
![fig2](https://github.com/user-attachments/assets/a4feffce-b4f4-461f-ac81-3efc5b3c4fcf)

After completing the "second forgetting task," `test old-1` refers to the model's performance on the test dataset from the "first forgetting task." This result should ideally get smaller, and the paper reports a value of 0. However, I cannot achieve even a comparable result.
![fig3](https://github.com/user-attachments/assets/474fbeba-2853-492e-88b8-68e5c3f3d33d)

After completing the "third forgetting task," `test old-2` refers to the model's performance on the test datasets from the "first two forgetting tasks."

## Anomalies in the ACC_o Metric 2

Additionally, I noticed that the `Test-forget-0 dataset`, used during the "first forgetting task," should be the same as the `test old-1 dataset` during the "second forgetting task." These two metrics should yield equivalent results for the model before learning task 2, which can also be found in the training log provided by the author.
![图片1](https://github.com/user-attachments/assets/286c6f5b-40fd-4c8f-973c-9688aadfbd67)

However, in my training log, the values of these two metrics are not the same, and I am very curious as to why this is the case.

![图片2](https://github.com/user-attachments/assets/2ea2d9d2-c8d1-478d-90d8-7ed7aab984d2)

**Comment by bjzhb666** [comment]
We have updated the latest code, please refer to it.
